tests/test_algorithm.py

```diff
@@ -2018,3 +2022,99 @@ def test_remove_data(self):
+    """
+    def setUp(self):
+        dt = pd.Timestamp('2015-01-05', tz='UTC')
+        self.env = TradingEnvironment()
+        ix = self.env.trading_days.get_loc(dt)
+
+        self.metadata = make_rotating_equity_info(
+            3, dt, trading_day, 2, 2,
+        )
+
+        # Make all start dates equal.
+        self.metadata['start_date'] = [self.metadata['start_date'][0]]*3
+
+        # Add an auto close date column, where liquidation occurs the day after
+        # the end date.
+        self.metadata['auto_close_date'] = \
+            self.metadata['end_date'] + trading_day
+
+        index_x = self.env.trading_days[ix:ix + 3]
+        data_x = pd.DataFrame(
+            data=[[1, 100], [2, 100], [3, 100]],
+            index=index_x,
+            columns=['price', 'volume'],
+        )
+        index_y = self.env.trading_days[ix:ix + 5]
+        data_y = pd.DataFrame(
+            data=[[4, 100], [5, 100], [6, 100], [7, 100], [8, 100]],
+            index=index_y,
+            columns=['price', 'volume'],
+        )
+        index_z = self.env.trading_days[ix:ix + 7]
+        data_z = pd.DataFrame(
+            data=[
+                [9, 100], [10, 100], [11, 100], [12, 100], [13, 100],
+                [14, 100], [15, 100],
+            ],
+            index=index_z,
+            columns=['price', 'volume'],
+        )
+
+        pan = pd.Panel({0: data_x, 1: data_y, 2: data_z})
+        self.source = DataPanelSource(pan)
+
+    def tearDown(self):
+        del self.env
+
+    def test_delisted_securities(self):
+        """
+        Make sure that after a security gets delisted, our portfolio holds the
+        correct number of securities and correct amount of cash.
+        """
+        self.env.write_data(equities_df=self.metadata)
+        algo = TestLiquidationAlgorithm(env=self.env)
+        algo.run(self.source)
+
+        expected_cash = [100000, 99830, 99830, 99860, 99860, 99940, 99940]
+        expected_num_positions = [0, 3, 3, 2, 2, 1, 1]
+        self.assertEqual(algo.cash, expected_cash)
+        self.assertEqual(algo.num_positions, expected_num_positions)
+
+    def test_delisted_securities_with_lag(self):
+        """
+        Test that an equity's auto close date functions properly when it is
+        set multiple days after the equity's end date.
+        """
+        # Add an auto close date column, where liquidation occurs 2 days after
+        # the end date.
+        metadata = self.metadata.copy()
+        metadata['auto_close_date'] = metadata['end_date'] + (2 * trading_day)
+
+        self.env.write_data(equities_df=metadata)
+        algo = TestLiquidationAlgorithm(env=self.env)
+        algo.run(self.source)
+
+        expected_cash = [100000, 99830, 99830, 99830, 99860, 99860, 99940]
+        expected_num_positions = [0, 3, 3, 3, 2, 2, 1]
+        self.assertEqual(algo.cash, expected_cash)
+        self.assertEqual(algo.num_positions, expected_num_positions)
+
+    def test_cancel_open_orders(self):
```

Comment:
My thinking here was that if you place an order for a stock the day before it delists, it will never get filled and under the current circumstances would remain open indefinitely, meaning at the end of the backtest you'd still have an open order for it. This tests that at the end of the backtest this is not the case. I will fix it though to make it clearer.